OctoKit/Issue.swift

```diff
@@ -3,75 +3,57 @@ import RequestKit
 
 // MARK: model
 
-public enum Openness: String {
+public enum Openness: String, Codable {
     case Open = "open"
     case Closed = "closed"
     case All = "all"
 }
 
-@objc open class Issue: NSObject {
-    open var id: Int
-    open var url: URL?
-    open var repositoryURL: URL?
-    open var labelsURL: URL?
-    open var commentsURL: URL?
-    open var eventsURL: URL?
-    open var htmlURL: URL?
+@objc open class Issue: NSObject, Codable {
+    @objc open private(set) var id: Int = -1
```

Comment:
Hey @pietbrauer, I'm working on porting my RequestKit based framework to Swift 4 as well, could you please explain why we need the extra words for Codable? Thanks!

Reply:
Also, `@objc` seems to be deprecated in Swift 4, is there a new alternative to `@objc`? 
Cheers.

Reply:
It is not deprecated, it just changed. Before if you marked a class as Objective-C available all its properties where also Objective-C available. In Swift 4 you have to mark them explicitly as Objective-C available so you can access them from Objective-C code.

Reply:
Got it, thanks!